Added docs for ExternalProject download during CMake step

audiofanatic opened the pull request:
Added content to the gtest and gmock build instructions with links to an
article explaining how to download googletest sources at CMake time
and then use `add_subdirectory()` to incorporate it directly into an
existing build. The preference for using CMake was made clearer in
the gmock readme and the advantage of building gtest and gmock
directly as part of an existing CMake build has been highlighted.

BillyDonahue commented on googletest/README.md:
```diff
@@ -80,13 +86,37 @@ using Visual Studio.
+project to download/copy the gtest sources as part of the configure
+step, then pull that downloaded gtest directory into the build using
+`add_subdirectory()`. This technique is relatively straightforward
+and is fully explained in
```
I'd like a quick example here so we're not completely dependent on the crascit article.

audiofanatic commented:
Hopefully that's about the level of example code you wanted to see. It should be enough for people to see how it's done without needing to read the article. There's a little more detail in the article explaining what's happening, but the essentials are there in the googletest readme now.

audiofanatic commented:
Sorry about the last commit, had to force push a correction to the commit message. I've updated the docs so that it is clearer about the other ways CMake can be used and to give less prominence to the Crascit article. I think it is more in keeping with the rest of the document's style now. The merging of #657 also meant I was able to update the example to be a bit cleaner.

audiofanatic commented:
Coming back to this after no activity/response for nearly a year. I've rebased and squashed the original set of commits to make it easier to see what has been changed (i.e. just the two .md files). I also slightly updated the instructions to include error checking on the two `execute_process()` calls. Is there anything more needed in order to get this merged?
